tests/L0/run_transformer/run_gpt_minimal_test.py

```diff
@@ -0,0 +1,172 @@
+import random
+import torch
+import traceback
+from typing import List
+from functools import partial
+from apex.transformer import tensor_parallel
+from apex.transformer import parallel_state
+from apex.transformer.tensor_parallel import vocab_parallel_cross_entropy
+from apex.transformer.tensor_parallel import model_parallel_cuda_manual_seed
+
+from apex.transformer.pipeline_parallel.utils import setup_microbatch_calculator
+from apex.transformer.pipeline_parallel.utils import average_losses_across_data_parallel_group
+from apex.transformer.pipeline_parallel.utils import get_ltor_masks_and_position_ids
+#from apex.transformer.pipeline_parallel.utils import update_num_microbatches
+from apex.transformer.pipeline_parallel.schedules.common import build_model
+from apex.transformer.pipeline_parallel.schedules.common import _get_params_for_weight_decay_optimization
+from apex.transformer.pipeline_parallel.schedules.fwd_bwd_pipelining_without_interleaving import forward_backward_pipelining_without_interleaving
+
+from apex.transformer.testing.standalone_gpt import post_language_model_processing, gpt_model_provider 
+from apex.transformer.testing import global_vars
+from apex.transformer.testing.commons import TEST_SUCCESS_MESSAGE
+from apex.transformer.testing.commons import initialize_distributed
+from apex.transformer.testing.commons import print_separator
+
+mode = None
+MANUAL_SEED = 42
+inds = None
+masks = None
+data_idx = 0
+MASK_PROB = 0.1
+EASY_MODE = False
+N_VOCAB = 128
+# download a public domain book as corpus
+def download_fancy_data():
+  import requests
+  response = requests.get('https://www.gutenberg.org/files/1342/1342-0.txt')
+  #response = requests.get('https://www.gutenberg.org/files/84/84-0.txt')
+  text = ' '.join(response.text.split())
+  encoded = text.encode('ascii', 'replace')
+  ints = [int(encoded[i]) for i in range(len(encoded))]
+  return torch.tensor(ints)
+
+# build a batch given sequence_len and batch size
+def generate_fancy_data_labels(sequence_len, batch_size):
+  global data_idx
+  global inds
+  global MANUAL_SEED
+  temps = list()
+  for i in range(batch_size):
+    if inds is None or data_idx >= len(inds):
+       # hack as use of RNG will fall out of sync due to pipelines being different
+      model_parallel_cuda_manual_seed(MANUAL_SEED)
+      inds = torch.randperm(effective_length, device='cuda')
+      MANUAL_SEED += 1
+      data_idx = 0
+    data_idx_ = data_idx
+    offset = inds[data_idx_] #* SEQUENCE_LEN
+    data_idx += 1
+    curr = fancy_data[offset:offset+sequence_len].clone().detach()

```

Comment:
I think the following suggestion is needed because `get_batch` function curtails the sequence by one element in [L.70](https://github.com/NVIDIA/apex/pull/1222/files#diff-c300b35d6e57bca85106f7aac069ff5856e5ee68966b305fbb9bb55aef5d3f20R70) & [L.71](https://github.com/NVIDIA/apex/pull/1222/files#diff-c300b35d6e57bca85106f7aac069ff5856e5ee68966b305fbb9bb55aef5d3f20R71), i.e. the "sequence length" dim of `tokens` and `labels` is unexpectedly `args.seq_length - 1`
```suggestion
    curr = fancy_data[offset:offset+sequence_len + 1].clone().detach()
```